src/canmatrix/canmatrix.py

```diff
@@ -1119,11 +1287,18 @@ def recalcDLC(self, strategy):
                 frame.size = math.ceil(maxBit / 8)
 
     def renameEcu(self, old, newName):
-        if type(old).__name__ == 'instance':
+        """Rename ECU in the Matrix. Update references in all Frames.
+
+        :param str or BoardUnit old: old name or ECU instance
+        :param str newName: new name
+        """
+        if type(old).__name__ == 'instance':  # accept BoardUnit instance
             pass
-        else:
+        else:  # or string
```

Comment:
Yeah, we need to stop doing this type guessing stuff.  I'd say either `renameEcuByBoardUnit()` and `renameEcuByName()` or we just check if old is in `self.boardUnits` and not then treat it as a string.  But, this is a separate issue.

Not to mention that when we do check type it should be `isinstance(this, That)`. :]  Looks like there are a few places to apply this.

Reply:
I fully agree

Reply:
Leaving the code untouched for further refactoring.